huaweicloud/services/acceptance/secmaster/resource_huaweicloud_secmaster_incident_test.go

```diff
@@ -0,0 +1,198 @@
+					resource.TestCheckResourceAttr(rName, "data_source.0.product_name", "hss"),
+					resource.TestCheckResourceAttr(rName, "data_source.0.source_type", "1"),
+					resource.TestCheckResourceAttr(rName, "first_occurrence_time", "2023-04-18T14:00:00.000+08:00"),
+					resource.TestCheckResourceAttr(rName, "last_occurrence_time", "2023-04-19T15:00:00.000+08:00"),
+					resource.TestCheckResourceAttr(rName, "verification_status", "Positive"),
+					resource.TestCheckResourceAttr(rName, "stage", "Detection and analysis"),
+					resource.TestCheckResourceAttr(rName, "debugging_data", "true"),
+					resource.TestCheckResourceAttr(rName, "labels", "test1,test2,test3"),
+				),
+			},
+			{
+				ResourceName:      rName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: testIncidentImportState(rName),
+				ImportStateVerifyIgnore: []string{
+					"updated_at",
+				},
+			},
+		},
+	})
+}
+
+func testIncident_basic(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_secmaster_incident" "test" {
+  workspace_id = "%s"
+  name         = "%s"
+  description  = "test incident"
+
+  type {
+    category      = "DDoS"
+    incident_type = "ACK Flood"
+  }
+
+  level  = "Tips"
+  status = "Open"
+
+  data_source {
+    product_feature = "hss"
+    product_name    = "hss"
+    source_type     = 1
+  }
+
+  first_occurrence_time = "2023-04-18T13:00:00.000+08:00"
+  last_occurrence_time  = "2023-04-19T14:00:00.000+08:00"
+  verification_status   = "Unknown"
+  stage                 = "Preparation"
+  debugging_data        = false
+  labels                = "test1,test2"
+}
+`, acceptance.HW_SECMASTER_WORKSPACE_ID, name)
+}
+
+func testIncident_update(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_secmaster_incident" "test" {
+  workspace_id = "%s"
+  name         = "%s"
+  description  = "test incident update"
+
+  type {
+    category      = "Web Attack"
+    incident_type = "Black IP"
+  }
+
+  level  = "Low"
+  status = "Block"
+
+  data_source {
+    product_feature = "hss"
+    product_name    = "hss"
+    source_type     = 1
+  }
+
+  first_occurrence_time = "2023-04-18T14:00:00.000+08:00"
+  last_occurrence_time  = "2023-04-19T15:00:00.000+08:00"
+  verification_status   = "Positive"
+  stage                 = "Detection and analysis"
+  debugging_data        = true
+  labels                = "test1,test2,test3"
+}
+`, acceptance.HW_SECMASTER_WORKSPACE_ID, name)
+}
+
+func testIncidentImportState(name string) resource.ImportStateIdFunc {
+	return func(s *terraform.State) (string, error) {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return "", fmt.Errorf("resource (%s) not found: %s", name, rs)
+		}
+		if rs.Primary.Attributes["workspace_id"] == "" {
+			return "", fmt.Errorf("attribute (workspace_id) of Resource (%s) not found: %s", name, rs)
+		}
+
+		return rs.Primary.Attributes["workspace_id"] + "/" + rs.Primary.ID, nil
```

Comment:
maybe `fmt.Sprintf("%s/%s", rs.Primary.Attributes["workspace_id"], rs.Primary.ID)` is better for reading code?

Reply:
sure